### PR TITLE
gwyddion 2.41

### PR DESCRIPTION
**Pull request by tuxu**
Update Gwyddion to 2.41. Also, add Python search path to prevent linking the
wrong library.

### CODE REVIEW

**Comment by DomT4** [comment]
```
        Error: 1 problem in 1 formula
gwyddion:
 * python modules have explicit framework links
These python extension modules were linked directly to a Python
framework binary. They should be linked with -undefined dynamic_lookup
instead of -lpython or -framework Python.
  /usr/local/Cellar/gwyddion/2.41/lib/python2.7/site-packages/gwy.so
```

**Comment by tuxu** [comment]
I should have checked the audit before. The check on explicit Python linkage was new to me.

File: Library/Formula/gwyddion.rb [hunk 1]
```diff
@@ -23,6 +22,12 @@ class Gwyddion < Formula
   depends_on "gtksourceview" if build.with? "python"
 
   def install
+    inreplace "configure", 'PYTHON_LIBS=-l$libpython',
+                           %(py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
+                           PYTHON_LIBS="-L${py_prefix}/lib -l$libpython")
+    inreplace "modules/pygwy/Makefile.in", '$(PYTHON_LIBS) $(PYGTK_LIBS) @GTK_LIBS@',
+                                           '-undefined dynamic_lookup $(PYGTK_LIBS) @GTK_LIBS@'
```
Comment (mikemcquaid) [inline Library/Formula/gwyddion.rb]: Can you please submit these patches to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (tuxu) [inline Library/Formula/gwyddion.rb]: Sure, will do!

**Comment by DomT4** [comment]
Ideally we want to get a patch upstream for the change to Python handling. @tdsmith will be able to sign off on whether we're content to carry these prior to them being merged upstream, I think it's okay but not the expert.

**Comment by tuxu** [comment]
I submitted a [patch upstream](http://sourceforge.net/p/gwyddion/mailman/message/34347458/). Hope this will be merged with the next release. I had far too many encounters with segfaults due to linking the wrong Python libs in the past.

@tdsmith, would you please consider merging this pull request now? Without these changes, Python support is broken in Gwyddion.

**Comment by tdsmith** [comment]
Looks great; pulled in a8eedc57. Thanks for your pull request! :tada: Homebrew thrives because of contributions like this one.

**Comment by DomT4** [comment]
@tdsmith Thanks Tim :heart:

**Pull request closed without merging**